pkg/controllers/management/multiclusterapp/enqueue.go

```diff
@@ -0,0 +1,31 @@
+package multiclusterapp
+
+import (
+	"context"
+	"github.com/rancher/rancher/pkg/namespace"
+	"github.com/rancher/types/apis/management.cattle.io/v3"
+	pv3 "github.com/rancher/types/apis/project.cattle.io/v3"
+	"github.com/rancher/types/config"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+func StartMCAppEnqueueController(ctx context.Context, management *config.ManagementContext) {
+	m := MCAppEnqueueController{
+		mcApps: management.Management.MultiClusterApps(""),
+	}
+	management.Project.Apps("").AddHandler(ctx, "management-mcapp-enqueue-controller", m.sync)
+}
+
+type MCAppEnqueueController struct {
+	mcApps v3.MultiClusterAppInterface
+}
+
+func (m *MCAppEnqueueController) sync(key string, app *pv3.App) (runtime.Object, error) {
+	if app == nil || app.DeletionTimestamp != nil {
+		return app, nil
+	}
+	if mcappName, ok := app.Labels[MultiClusterAppIDSelector]; ok {
```

Comment:
do we have to enqueue on the app removal?

Reply:
at least for the situation when app is not nil

Reply:
We don't allow apps controlled by mc app to be removed by user, so I was thinking this would be called only when mc app is actually deleted. I'm not 100% sure if the sync on mc app is required then, but enqueued just in case